src/Servers/Kestrel/Core/src/Internal/Http3/Http3HeadersEnumerator.cs

```diff
@@ -147,7 +148,89 @@ public void Dispose()
 
     internal static int GetResponseHeaderStaticTableId(KnownHeaderType responseHeaderType)
     {
-        // Not Implemented
-        return -1;
+        // Not every header in the QPACK static table is known.
+        // These are missing from this test and the full header name is written.
+        // Missing:
+        // - link
+        // - location
+        // - strict-transport-security
+        // - x-content-type-options
+        // - x-xss-protection
+        // - content-security-policy
+        // - early-data
+        // - expect-ct
+        // - purpose
+        // - timing-allow-origin
+        // - x-forwarded-for
+        // - x-frame-options
+        switch (responseHeaderType)
+        {
+            case KnownHeaderType.Age:
+                return H3StaticTable.Age0;
+            case KnownHeaderType.ContentLength:
+                return H3StaticTable.ContentLength0;
+            case KnownHeaderType.Date:
+                return H3StaticTable.Date;
+            case KnownHeaderType.Cookie:
```

Comment:
We should consider adding the un-"known" response headers given the methodology described in the [quickwg QPACK static table appendix](https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#name-static-table-2).

I also think the following headers only make sense on the request: Cookie, Authorization, UserAgent, Referer, Origin, IfModifiedSince, IfNoneMatch, Method, Accept, AcceptEncoding, AcceptLanguage, Range, IfRange and UpgradeInsecureRequests

Do we think there would be much value in trying to use the static header values too?

Reply:
I didn't add more known headers because there is a limit to how many we can have. Eventually, the bit flags will be too large for 64bits.

I will look at using the static table for values in a future PR. It will require greater changes.

Reply:
> I also think the following headers only make sense on the request: Cookie, Authorization, UserAgent, Referer, Origin, IfModifiedSince, IfNoneMatch, Method, Accept, AcceptEncoding, AcceptLanguage, Range, IfRange and UpgradeInsecureRequests

Yes, remove request headers from this list.

Reply:
For HPACK, we have the complete list: https://github.com/dotnet/aspnetcore/blob/465f9294d0b9f5403ab1a25c42ca9cf75e3cdeae/src/Servers/Kestrel/Core/src/Internal/Http2/Http2HeadersEnumerator.cs#L149-L203

Do you want to trim the HPACK list as well? Which items?

Or leave both HPACK and QPACK lists as is?

Reply:
Oh, ok, leave it.

Reply:
It looks to me that Http2HeadersEnumerator.GetResponseHeaderStaticTableId() only handles response headers. I don't see request headers like Cookie, Authorization or any of the rest in there.

Reply:
You're right. I'll make the change to remove them in another PR.